tests/contrib/mariadb/test_mariadb.py

```diff
@@ -0,0 +1,386 @@
+    proc = "sp_sum"
+    data = (40, 2, None)
+    cursor.callproc(proc, data)
+
+    spans = tracer.pop()
+    assert spans, spans
+
+    # number of spans depends on mariadb implementation details,
+    # typically, internal calls to execute, but at least we
+    # can expect the last closed span to be our proc.
+    span = spans[len(spans) - 1]
+    assert_is_measured(span)
+    assert span.service == "mariadb"
+    assert span.name == "mariadb.query"
+    assert span.span_type == "sql"
+    assert span.error == 0
+    assert span.get_metric("out.port") == 3306
+    assert_dict_issuperset(
+        span.meta,
+        {
+            "out.host": u"127.0.0.1",
+            "db.name": u"test",
+            "db.user": u"test",
+        },
+    )
+    assert span.get_tag("mariadb.query") is None
+
+
+def test_commit(connection, tracer):
+    connection.commit()
+    spans = tracer.pop()
+    assert len(spans) == 1
+    span = spans[0]
+    assert span.service == "mariadb"
+    assert span.name == "mariadb.connection.commit"
+
+
+def test_rollback(connection, tracer):
+    connection.rollback()
+    spans = tracer.pop()
+    assert len(spans) == 1
+    span = spans[0]
+    assert span.service == "mariadb"
+    assert span.name == "mariadb.connection.rollback"
+
+
+def test_analytics_default(connection, tracer):
+    cursor = connection.cursor()
+    cursor.execute("SELECT 1")
+    rows = cursor.fetchall()
+    assert len(rows) == 1
+    spans = tracer.pop()
+    assert len(spans) == 1
+    span = spans[0]
+    assert span.get_metric(ANALYTICS_SAMPLE_RATE_KEY) is None
+
+
+def test_analytics_with_rate(connection, tracer):
+    with override_config("mariadb", dict(analytics_enabled=True, analytics_sample_rate=0.5)):
+        cursor = connection.cursor()
+        cursor.execute("SELECT 1")
+        rows = cursor.fetchall()
+        assert len(rows) == 1
+        spans = tracer.pop()
+        assert len(spans) == 1
+        span = spans[0]
+        assert span.get_metric(ANALYTICS_SAMPLE_RATE_KEY) == 0.5
+
+
+def test_analytics_without_rate(connection, tracer):
+    with override_config("mariadb", dict(analytics_enabled=True)):
+        cursor = connection.cursor()
+        cursor.execute("SELECT 1")
+        rows = cursor.fetchall()
+        assert len(rows) == 1
+        spans = tracer.pop()
+
+        assert len(spans) == 1
+
+        span = spans[0]
+        assert span.get_metric(ANALYTICS_SAMPLE_RATE_KEY) == 1.0
+
+
+@pytest.mark.parametrize(
+    "service_env_key,service_env_value", [("DD_SERVICE", "mysvc"), ("DD_MARIADB_SERVICE", "mysvc")]
+)
+def test_user_specified_service_snapshot(run_python_code_in_subprocess, service_env_key, service_env_value):
+    """
+    When a user specifies a service for the app
+        The mariadb integration should not use it.
+    """
+
+    @snapshot(
+        async_mode=False,
+        token_override="tests.contrib.mariadb.test_mariadb.test_user_specified_service_snapshot_{}_{}".format(
```

Comment:
:+1: this workaround is required since the `snapshot` decorator isn't compatible with `pytest.mark.parametrize` right now unfortunately.

@Yun-Kim not sure if you've happened to dig into some of the pytest internals yet but if so do you think it'd be much work to make `snapshot` compatible with parametrize?

Reply:
Sorry, just saw this comment. I haven't dove into `snapshot` with `parameterize` yet, but I'll look into it and let you know if there's an easy compatibility fix.

Reply:
No rush or anything - just something to think about! We don't need to block this PR on it but we probably want to have them be compatible before we see this pattern 1000 times in the project 😆